### PR TITLE
[CSL-2135] Print difference between tip slot and current slot

### PR DESCRIPTION
**Pull request by kantp**
Part of the backport of #2213.

### CODE REVIEW

File: node/src/Pos/Block/Worker.hs [hunk 1]
```diff
@@ -97,11 +97,21 @@ blkOnNewSlot =
...
+    -> m ()
 blockCreator (slotId@SlotId {..}) sendActions = do
 
+    -- Print difference between tip slot and current slot.
+    -- TODO: probably not the best place, but it's not the biggest concern now.
+    tipHeader <- DB.getTipHeader @ssc
+    let tipSlot = epochOrSlotToSlot (getEpochOrSlot tipHeader)
+    let slotDiff = flattenSlotId slotId - flattenSlotId tipSlot
+    logInfo $ sformat ("Difference between current slot and tip slot is: "%int)
+              slotDiff
```
Comment (dcoutts) [inline node/src/Pos/Block/Worker.hs]: Slightly better would be to name the whole thing and put it in the where clause, so then it doesn't clutter things up so much, and we know that `tipHeader` is not in scope and possibly used below. I suggest `logHowManySlotsBehind`.

**Comment by dcoutts** [comment]
Otherwise this is fine, will approve.

**Comment by adinapoli-iohk** [comment]
As @kantp is unavailable this morning, will try to apply the suggested fixes to speed things up 😉  I hope it's fine with you, Philipp!

**Comment by adinapoli-iohk** [comment]
@kantp , @dcoutts ready for re-review!

**Comment by kantp** [comment]
Thanks, @adinapoli-iohk!

File: node/src/Pos/Block/Worker.hs [hunk 2]
```diff
@@ -97,11 +97,18 @@ blkOnNewSlot =
...
+blockCreator ::
+       forall ssc ctx m. WorkMode ssc ctx m
+    => SlotId
+    -> SendActions m
+    -> m ()
 blockCreator (slotId@SlotId {..}) sendActions = do
 
+    -- Print difference between tip slot and current slot.
+    -- TODO: probably not the best place, but it's not the biggest concern now.
+    tipHeader <- DB.getTipHeader @ssc
```
Comment (dcoutts) [inline node/src/Pos/Block/Worker.hs]: My suggestion had been to move this into the where clause too, to show that tipHeader is not used anywhere else later in the scope.

Comment (adinapoli-iohk) [reply node/src/Pos/Block/Worker.hs]: Ah, sorry @dcoutts , I understood something different, my bad 🙀

**Pull request merged**